src/pages/LEVANTE/RegisterUsers.vue

```diff
@@ -0,0 +1,493 @@
+          </PvDataTable>
+        </div>
+      </div>
+
+      <div v-if="returnedData.length">
+        <PvButton label="Download Registered Users" @click="addAccountToCSV(returnedData)" />
+      </div>
+    </section>
+  </main>
+</template>
+<script setup>
+import { ref, toRaw } from 'vue';
+import { csvFileToJson } from '@/helpers';
+import _cloneDeep from 'lodash/cloneDeep';
+import _compact from 'lodash/compact';
+import _forEach from 'lodash/forEach';
+import _isEmpty from 'lodash/isEmpty';
+import _startCase from 'lodash/startCase';
+import { useAuthStore } from '@/store/auth';
+import { useToast } from 'primevue/usetoast';
+import { pluralizeFirestoreCollection } from '@/helpers';
+import { fetchOrgByName } from '@/helpers/query/orgs';
+
+const authStore = useAuthStore();
+const toast = useToast();
+const isFileUploaded = ref(false);
+const rawUserFile = ref({});
+const returnedData = ref([]);
+
+// Primary Table & Dropdown refs
+const dataTable = ref();
+const tableColumns = ref([]);
+const dropdown_model = ref({});
+const dropdown_options = ref([
+  {
+    label: 'Required',
+    items: [
+      { label: 'ID', value: 'id' },
+      { label: 'User Type', value: 'userType' },
+      { label: 'Name', value: 'name' },
+      { label: 'Child ID', value: 'childId' },
+      { label: 'Parent ID', value: 'parentId' },
+      { label: 'Teacher ID', value: 'teacherId' },
+      { label: 'Month', value: 'month' },
+      { label: 'Year', value: 'year' },
+      { label: 'Group', value: 'group' },
+    ],
+  },
+]);
+const requiredFields = ['id', 'userType', 'name', 'childId', 'parentId', 'teacherId', 'month', 'year', 'group'];
+
+// Error Users Table refs
+const errorTable = ref();
+const errorUsers = ref([]);
+const errorUserColumns = ref([]);
+const errorMessage = ref('');
+const showErrorTable = ref(false);
+
+const activeSubmit = ref(false);
+
+// Functions supporting the uploader
+const onFileUpload = async (event) => {
+  rawUserFile.value = await csvFileToJson(event.files[0]);
+  tableColumns.value = generateColumns(toRaw(rawUserFile.value[0]));
+  populateDropdown(tableColumns.value);
+  isFileUploaded.value = true;
+  toast.add({ severity: 'success', summary: 'Success', detail: 'File Successfully Uploaded', life: 3000 });
+};
+
+function populateDropdown(columns) {
+  _forEach(columns, (col) => {
+    dropdown_model.value[col.field] = '';
+  });
+}
+
+function generateColumns(rawJson) {
+  let columns = [];
+  const columnValues = Object.keys(rawJson);
+  _forEach(columnValues, (col) => {
+    let dataType = typeof rawJson[col];
+    if (dataType === 'object') {
+      if (rawJson[col] instanceof Date) dataType = 'date';
+    }
+    columns.push({
+      field: col,
+      header: _startCase(col),
+      dataType: dataType,
+    });
+  });
+  return columns;
+}
+
+function getKeyByValue(object, value) {
+  return Object.keys(object).find((key) => object[key] === value);
+}
+
+function areArraysEqual(arr1, arr2) {
```

Comment:
For simplicity, this function could be substituted with [lodash's isEqual](https://lodash.com/docs/4.17.15#isEqual)